x-pack/plugins/apm/server/lib/services/get_services/get_service_transaction_stats.ts

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  TRANSACTION_PAGE_LOAD,
+  TRANSACTION_REQUEST,
+} from '../../../../common/transaction_types';
+import { AgentName } from '../../../../typings/es_schemas/ui/fields/agent';
+import { EventOutcome } from '../../../../common/event_outcome';
+import {
+  AGENT_NAME,
+  EVENT_OUTCOME,
+  SERVICE_ENVIRONMENT,
+  SERVICE_NAME,
+  TRANSACTION_TYPE,
+} from '../../../../common/elasticsearch_fieldnames';
+import { rangeFilter } from '../../../../common/utils/range_filter';
+import {
+  getProcessorEventForAggregatedTransactions,
+  getTransactionDurationFieldForAggregatedTransactions,
+} from '../../helpers/aggregated_transactions';
+import { getBucketSize } from '../../helpers/get_bucket_size';
+import { ServicesItemsSetup } from './get_services_items';
+
+interface AggregationParams {
+  setup: ServicesItemsSetup;
+  searchAggregatedTransactions: boolean;
+}
+
+const MAX_NUMBER_OF_SERVICES = 500;
+
+export async function getServiceTransactionStats({
+  setup,
+  searchAggregatedTransactions,
+}: AggregationParams) {
+  const { apmEventClient, start, end, esFilter } = setup;
+
+  const count = {
+    value_count: {
+      field: getTransactionDurationFieldForAggregatedTransactions(
+        searchAggregatedTransactions
+      ),
+    },
+  };
+
+  const metrics = {
+    count,
+    avg_duration: {
+      avg: {
+        field: getTransactionDurationFieldForAggregatedTransactions(
+          searchAggregatedTransactions
+        ),
+      },
+    },
+    failure: {
+      filter: { term: { [EVENT_OUTCOME]: EventOutcome.failure } },
+      aggs: { count },
+    },
+    successful: {
+      filter: { term: { [EVENT_OUTCOME]: EventOutcome.success } },
+      aggs: { count },
+    },
+  };
+
+  const response = await apmEventClient.search({
+    apm: {
+      events: [
+        getProcessorEventForAggregatedTransactions(
+          searchAggregatedTransactions
+        ),
+      ],
+    },
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [{ range: rangeFilter(start, end) }, ...esFilter],
```

Comment:
I think this needs an additional `getDocumentTypeFilterForAggregatedTransactions`, otherwise we are aggregating on all metric documents if transaction metrics are enabled.